### ISSUE TITLE
Rules for dervative of asec and acsc with complex input incorrect? 

### ISSUE THREAD

**Issue by oxinabox** [JuliaDiff/ChainRules.jl#86]
Found when doing #76 

### `asec`
```
x = 1.6 - 0.8im
(central_fdm(5, 1))(asec, x) = 0.16078185380606091 + 0.29837305540037473im
(last(frule(asec, x)))(1) = 0.27724414876919296 + 0.19496914288010564im
```
[WolframAlpha says](https://www.wolframalpha.com/input/?i=derivative+of+asec%28x%29+at+x%3D%281.6+-+0.8i%29) `0.160782 + 0.298373 i`
Argreeing with the `central_fdm`

### `acsc`
```
x = 1.6 - 0.8im
(central_fdm(5, 1))(acsc, x) = -0.16078185380578297 - 0.29837305540024617im
(last(frule(acsc, x)))(1) = -0.27724414876919296 - 0.19496914288010564im
```
[WolframAlpha says](https://www.wolframalpha.com/input/?i=derivative+of+acsc%28x%29+at+x%3D%281.6+-+0.8i%29) `-0.160782 - 0.298373 i`
Argreeing with the `central_fdm`

@simeonschaub  do you know what is going on here?
It has been years since I did complex calculus

**Comment by oxinabox** [comment]
Investigating further.
This is basically because we use a simplification of the derviative that only works for real numbers.
So it is just wrong in the general case.

**Comment by oxinabox** [comment]
I have a fix, and will include it in my PR fro #76
